src/Realm.C

```diff
@@ -921,26 +921,15 @@ void
 Realm::setup_nodal_fields()
 {
 #ifdef NALU_USES_HYPRE
-  hypreGlobalId_ = &(meta_data().declare_field<HypreIDFieldType>(
-    stk::topology::NODE_RANK, "hypre_global_id"));
+  fieldManager_->register_field("hypre_global_id", meta_data().get_parts());
 #endif
-  tpetGlobalId_ = &(meta_data().declare_field<TpetIDFieldType>(
-    stk::topology::NODE_RANK, "tpet_global_id"));
+  fieldManager_->register_field("tpet_global_id", meta_data().get_parts());
```

Comment:
This is pre-existing before your commit, but tpet_global_id should be conditional on NALU_USES_TRILINOS_SOLVERS